apache/metron
# METRON-2292: [UI] Manual query can't be saved on the Alerts UI

## Contributor Comments

Link to the ASF Jira: https://issues.apache.org/jira/browse/METRON-2292

The problem
=========

When users switch to manual query mode and type something in the search field, it doesn't count when they want to save the search. However the manual mode is visible on the screen, it always saves the recent query built by the query builder (if it's anything else but '*')

Testing
=====

1. Go to the alerts-ui
2. Switch to manual query mode
3. Type something in the search field
4. Click on the search button
5. Click on the "Save" search button (floppy disc icon on the right of the magnifier)
6. Open the saved searches pane (Click the button on the left of the search field)
7. Choose your saved search from the list
8. Please try to choose it from the recent searches list to make sure it works properly.

Repeat the same with a saved search built by the query builder. The app should turn into the proper filtering mode based on the filtering mode of the saved search.

It's worth to refresh the browser between steps randomly just to make sure it's persisted properly. Since we only store saved and recent searches in the local storages, I recommend you to flush it before testing.

====

During testing, I've realised that the recent search means recent saved search. It was a bit confusing for the first time. I expected the recent search to be my latest search and not the recent saved search from the list below. Anyone else who thinks it might be confusing and misleading for the users?


## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```


#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
